class-udm-updater.php

```diff
@@ -930,5 +948,55 @@ protected function get_plugin_data($key = '') {
 		}
 		return false;
 	}
+
+	/**
+	 * Remove the use of auto_update index and its value from the *_updater_options option/meta (single/multisite) and change it to auto_update_plugins site option, which is also used by WordPress's core since version 5.5
+	 * This needs to be done in order to maintain auto-updates compatibility between WordPress and UDM and to synchronise the auto-updates setting for both
+	 */
+	protected function replace_auto_update_option() {
+		$options = $this->get_option($this->option_name);
+		if (!is_array($options)) $options = array();
+		$old_setting_value = isset($options['auto_update']) ? $options['auto_update'] : '';
+		$new_setting_value = $this->get_option('auto_update_plugins');
+		if (!is_array($new_setting_value)) $new_setting_value = array();
+		if (!empty($old_setting_value)) $new_setting_value[] = $this->relative_plugin_file;
+		$new_setting_value = array_unique($new_setting_value);
+		unset($options['auto_update']);
+		$options['run_replace_auto_update_option_once'] = true;
+		$this->update_option($this->option_name, $options);
+		$this->update_option('auto_update_plugins', $new_setting_value);
+	}
+
+	/**
+	 * Enable automatic updates for UDM
+	 */
+	protected function enable_automatic_updates() {
+		$auto_update_plugins = $this->get_option('auto_update_plugins');
+		if (!is_array($auto_update_plugins)) $auto_update_plugins = array();
+		$auto_update_plugins[] = $this->relative_plugin_file;
+		$auto_update_plugins = array_unique($auto_update_plugins);
+		$this->update_option('auto_update_plugins', $auto_update_plugins);
+	}
+
+	/**
+	 * Disable automatic updates for UDM
+	 */
+	protected function disable_automatic_updates() {
+		$auto_update_plugins = $this->get_option('auto_update_plugins');
+		if (!is_array($auto_update_plugins)) $auto_update_plugins = array();
+		$auto_update_plugins = array_diff($auto_update_plugins, array($this->relative_plugin_file));
+		$this->update_option('auto_update_plugins', $auto_update_plugins);
+	}
+
+	/**
+	 * Check whether the automatic-updates is set for UDM
+	 *
+	 * @return Boolean True if set, false otherwise
+	 */
+	protected function is_automatic_updates() {
```

Comment:
@andtown The name doesn't really work gramatically. Please call it "is_automatic_updating_enabled".